src/g_game.c

```diff
@@ -756,7 +756,7 @@ void G_BuildTiccmd(ticcmd_t* cmd)
         M_InputGameActive(input_weapon6) && gamemode != shareware ? wp_plasma :
         M_InputGameActive(input_weapon7) && gamemode != shareware ? wp_bfg :
         M_InputGameActive(input_weapon8) ? wp_chainsaw :
-        M_InputGameActive(input_weapon9) && !demo_compatibility && have_ssg ? wp_supershotgun :
+        M_InputGameActive(input_weapon9) && have_ssg ? wp_supershotgun :
```

Comment:
- Complevel 2. Press "9". It selects the fist. It should do nothing.
- Complevel 2. Pick up a chainsaw. Press "9". It selects the chainsaw. It should do nothing.

Reply:
Yes, that's because we have to use the original `BT_WEAPONMASK_OLD` in Vanilla complevel and this erases out the bit that is used to indicate a change to SSG.